docs/source/basic/faq.rst

```diff
@@ -0,0 +1,82 @@
+
+If you need certain operations supported for your model, please let us know by creating an issue. We plan to get to all of them (with the exception of any :doc:`sharp edges <sharp_edges>`), but your issues help us prioritize which to do first.
+
+There are potentially any number of other problems which could arise. Some of the problems are known, some may not be. Check out the :doc:`sharp edges <sharp_edges>` page. If what you're seeing still doesn't make sense, let us know by creating an issue.
+
+
+=======================================
+5. Does Thunder support dynamic shapes?
+=======================================
+
+No, not at the moment. However, we're actively working on it.
+
+Meta functions operate on the exact shapes of the tensor proxies that pass through them. This is a limitation of the current implementation, and we plan to incorporate dynamic shapes in the future. If you have relevant experience experience with this problem in pytorch, or you are interested in helping us implement it, please let us know by creating an issue or reaching out.
+
+
+================================================================
+6. Does Thunder support inplace operations?
+================================================================
+
+Not at the moment. Implementing inplace operations would require tracking which tensors in a trace have been modified by operations in our optimization passes, which currently we represent as purely functional. All deep learning compiler frameworks have to deal with the problem of tensor aliasing in some way. The way we've chosen for now is to pretend that the problem doesn't exist.
+
+The common solution is to represent programs in `SSA form <https://en.wikipedia.org/wiki/Static_single-assignment_form>`_, or do some form of SSA-inspired variable renaming, but SSA is a much less understandable representation than a list of symbols in a trace. Switching to SSA would also complicate optimization passes, and require rewriting many of them to handle these aliasing rules.
+
+There also exists the problem that some backends, like Triton and nvfuser, or a potential Jax backend, are not actually good at handling inplace operations either. They, like Thunder, expect functional programs, and if we simply decide to forward these operations to the executor, not every executor would be able to implement every operation. For a simple torch operator like ``x.add_(y)``, this is unsatisfactory to us. We would ideally like to be able to convert instances of ``add_()`` to ``add()`` for executors. But currently, every intermediate trace needs to be executable and `add_()` would not be a supported bound symbol before it gets replaced by `add()` later.
```

Comment:
Again, I want to thank you for doing this. It's a very important and often thankless task, plus it's bound to generate a bunch of word-nitting requests... like the one I'm about to make ;-)

> There also exists the problem that some backends, like Triton and nvfuser, or a potential Jax backend, are not actually good at handling inplace operations either.

There's potential for those who don't know us to wonder if we're trying to be snarky and taking stabs at other projects. Plus if one of these backends added SSA overnight someday, they might be disheartened to hear us still claiming issues in their SW.

If you agree, one way to would be to just simplify out the calling out of backends, e.g.:

 > Some backend executors do not support in-place operations.

It omits a lot of details, but those details are not needed to make the point needed for the rest of the paragraph.

Reply:
Funny enough, given the number of times I've been thanked, I don't think it's a thankless task anymore :)

You're right though, my bad. Any technical elaboration in the Thunder docs should probably only discuss Thunder, and obviously we're big fans of anything that makes models faster. That's sort of the whole point of the project, after all. Fixed.